.github/workflows/setup-tofu.yml

```diff
@@ -0,0 +1,184 @@
+name: 'Setup OpenTofu'
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  tofu-versions:
+    name: 'OpenTofu Versions'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        tofu-versions: [1.6.0-alpha1, latest]
+        tofu-wrapper: [true, false]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
```

Comment:
I don't see much of an advantage of using a specific commit here over just `v4` 

moving to v4 will also bring in the new 4.1.0 https://github.com/actions/checkout/releases/tag/v4.1.0 right now